Shutdown behavior (#135)
GameFabric/docs: src/multiplayer-servers/getting-started/vessel-shutdown-behavior.md

# Vessel Shutdown Behavior

To perform maintenance on long-running vessels without interfering with active game servers, the game server should respond to shutdown hints provided via annotations.

## Allocated Vessels

If shutdown or restart is required for an already allocated Vessel, the following two annotations will be added to each affected Vessel:

`g8c.io/shutdown-reason` with one of the following reasons:
- `UserInitiated`
- `SpecChange`
- `Maintenance`

and

`g8c.io/shutdown-timestamp` with a timestamp in the format of "2006-01-02T15:04:05Z07:00"

The timestamp will be `now` + the time configured for each reason via: `Settings`->`Advanced`->`Shutdown Notification`

::: info Multiple shutdown reasons
* The annotations are set once and never updated

> * they will disappear with the shutdown of the Vessel

Reviewer: I think it is clearer here to talk about the Game Server, not the Vessel. It alludes to the idea that they will not be removed if the Vessel is restarted, which is not true.
```suggestion
* they will disappear with the shutdown of the Game Server
```

> So, if there is a `SpecChange` with a higher wait period than a following `UserInitiated` shutdown, GameFabric will still wait the full `SpecChange period` before shutting it down

Reviewer: ```suggestion
So, if there is a `SpecChange` with a higher wait period than a following `UserInitiated` shutdown, GameFabric will still wait the full `SpecChange period` before shutting it down.
```

> it can of course being shutdown early by the game server itself responding to the annotations or leaving `Allocated` after all players left

Reviewer: ```suggestion
It can of course being shutdown early by the game server itself responding to the annotations or leaving `Allocated` after all players left.
```
